### PR TITLE
Provide configuration option "--maintainability-index-without-comments".

### PR DESCRIPTION
**Pull request by niconoe-**
Gives new functionnalities mentionned into #325

### CODE REVIEW

File: src/Hal/Report/Html/template/js/graph-maintainability.js [hunk 1]
```diff
@@ -37,31 +37,48 @@ function chartMaintainability() {
...
+            } else {
+                var text = '<strong>' + d.name + '</strong>'
+                    + "<br />Cyclomatic Complexity : " + d.ccn
+                    + "<br />Maintainability Index: " + d.mi;
+            }
             d3.select('.tooltip').html(text);
-            d3.select(".tooltip").style("opacity", 1);
+            d3.select(".tooltip")
+                .style("opacity", 1)
+                .style("z-index", 1);
```
Comment (niconoe-) [inline src/Hal/Report/Html/template/js/graph-maintainability.js]: Added "z-index" to fix issue moseover with the tooltip (tooltip was hidding the buubles).

File: src/Hal/Application/Config/Validator.php [hunk 2]
```diff
@@ -56,20 +56,21 @@ public function help()
...
-    --version                           Display current version
+    --exclude=<directory>                       List of directories to exclude, separated by a comma (,)
+    --extensions=<php,inc>                      List of extensions to parse, separated by a comma (,)
+    --report-html=<directory>                   Folder where report HTML will be generated
+    --report-csv=<file>                         File where report CSV will be generated
+    --report-json=<file>                        File where report Json will be generated
+    --report-violations=<file>                  File where XML violations report will be generated
+    --git[=</path/to/git_binary>]               Perform analyses based on Git History (default binary path: "git")
+    --junit[=</path/to/junit.xml>]              Evaluates metrics according to JUnit logs
+    --maintainability-index-without-comments    Ignore comments when calculating Maintainability Index
```
Comment (Halleck45) [inline src/Hal/Application/Config/Validator.php]: I'm not in favor of adding arguments for metrics.

What do you think about :
- add a checkbox "do not care of code comments" behind the graph ? 
- add new metric "maintability index without comments" ?

?

**Comment by gmponos** [comment]
> add new metric "maintability index without comments" ?
Ah.. I like this

**Comment by niconoe-** [comment]
> I'm not in favor of adding arguments for metrics. Otherwise, we'll have tons of specific arguments, and tool will be harder to use.
> 
> What do you think about :
> 
> * add a checkbox "do not care of code comments" behind the graph ? When checked, chart is updatd with the other metric
> * add new metric "maintability index without comments" ?
> 
> ?

I'm not a fan of the checkbox onto the graph because this is not only about the bubble graph: I also changed the MI displayed in the badges next to the Google Page Rank table. A more global usage of this "checkbox" is required then, but it will just move the complexity of having lots of metric configuration into having too many information displayed in the report.

That's the reason I choose the idea of having a configuration argument. Of course it brings higher complexity in the phpMetrics application, but I thought it could bring simplicity to the end user by not overloding him/her with unwanted metrics values.

I could easily change this if you do not want to keep this as a configuration, but we should discuss about the format in that case. More globally, we should agree on how can we deal with new metrics, and new way of reporting some of them.


Finally, I don't understand your question about "add new metric "maintability index without comments" ?": this metric already exists, even if only displayed in the "all.html" page.

**Comment by niconoe-** [comment]
In order to solve arguments, I decided to follow the advise of @Halleck45 and remove the option. I set a javascript (native) carousel in order to display the 2 bubble graphs (MI and MIwoC). I also change the Google Page rank panel to display both information in the badges.

Like I said, I was not a fan of the checkbox on the main page because I feel like it could reduce the UX. I think the carousel is better and all informations are accessible without clicking anywhere.

Is it an acceptable PR for you?
Thanks a lot.

**Comment by Halleck45** [comment]
Thanks @niconoe- !!

**Pull request merged**